linode/manager
# change: [M3-6522] - Select components stories cleanup

## Description 📝
Relatively small tech debt PR that does not affect the client facing code.

## Changes  🔄
- Remove `.mdx` story in favor of a `.tsx` one
- Mark 'Select/EnhancedSelect' as deprecated to avoid new usage
- Add new story for `<RegionSelect />`
- Group all "select" stories in storybook

## Preview 📷

![Screenshot 2023-11-14 at 15 39 11](https://github.com/linode/manager/assets/130582365/4b254253-fb98-4651-909e-448cc1cf8150)

## How to test 🧪

### Prerequisites
- Pull code locally and run `yarn storybook`

### Verification steps 
- Navigate to http://localhost:6006/?path=/docs/components-selects-enhanced-select--documentation and confirm changes & behavior of RegionSelect and EnhancedSelect stories

## As an Author I have considered 🤔

*Check all that apply*
